cfme/tests/cloud/test_cloud_events.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+"""This module tests only cloud specific events"""
+import pytest
+import re
+
+from cfme.common.vm import VM
+from utils import testgen
+from utils.events import EventBuilder
+from utils.generators import random_vm_name
+
+
+pytest_generate_tests = testgen.generate(testgen.cloud_providers, scope="module")
+pytestmark = [
+    pytest.mark.tier(3),
+    pytest.mark.usefixtures('uses_cloud_providers'),
+]
+
+
+@pytest.mark.uncollectif(lambda provider: provider.type != 'azure')
+def test_manage_nsg_group(provider, register_event):
+    """
+    tests that create/remove azure network security groups events are received and parsed by CFME
+    """
+
+    nsg_name = random_vm_name(context='nsg')
+    resource_group = provider.data['provisioning']['resource_group']
+
+    if not provider.exists:
+        provider.create()
+
+    # registering add/remove network security group events
+    # we need to check raw data by regexps, since many azure events aren't parsed by CFME yet
+    builder = EventBuilder()
+    fd_regexp = '^\s*resourceId:.*?{nsg}.*?^\s*status:.*?^\s*value:\s*{stat}.*?^' \
+                '\s*subStatus:.*?^\s*value:\s*{sstat}'
+
+    def add_cmp(_, y):
+        bool(re.search(fd_regexp.format(nsg=nsg_name, stat='Accepted',
+                                        sstat='Created'), y, re.M | re.U | re.S))
+    fd_add_attr = {'full_data': 'will be ignored',
+                   'cmp_func': add_cmp}
+
+    def rm_cmp(_, y):
+        bool(re.search(fd_regexp.format(nsg=nsg_name, stat='Succeeded',
+                                        sstat=' '), y, re.M | re.U | re.S))
+    fd_rm_attr = {'full_data': 'will be ignored',
+                  'cmp_func': rm_cmp}
+
+    add_event = builder.new_event(fd_add_attr, source='AZURE',
+                                  event_type='networkSecurityGroups_write_EndRequest')
+    register_event(add_event)
+
+    remove_event = builder.new_event(fd_rm_attr, source='AZURE',
+                                     event_type='networkSecurityGroups_delete_EndRequest')
+    register_event(remove_event)
+
+    # creating and removing network security group
+    provider.mgmt.create_netsec_group(nsg_name, resource_group)
+    provider.mgmt.remove_netsec_group(nsg_name, resource_group)
```

Comment:
register_event checks all events at the end of the tests and raises appropriate assert errors.
It uses soft_assert fixture. Actually, the error will be emmited in test's teardown because there is no other way to use soft_assert in other fixtures like register_event. But somehow or other it shouldn't influence tests.